File: acapella_api/app/Models/Choir.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Choir extends Model
{
    use SoftDeletes;

    protected $guarded = ['id'];

    protected $fillable = [
        'name',
        'location',
        'bio',
        'image_path',
        'is_verified',
        'user_id',
    ];

    protected function casts(): array
    {
        return [
            'is_verified' => 'boolean',
        ];
    }

    // ── Relationships ─────────────────────────────────────────────────────────

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function albums(): HasMany
    {
        return $this->hasMany(Album::class);
    }

    public function tracks(): HasMany
    {
        return $this->hasMany(Track::class);
    }

    // ── Scopes ───────────────────────────────────────────────────────────────

    public function scopeVerified($query)
    {
        return $query->where('is_verified', true);
    }

    public function scopeManagedBy($query, int $userId)
    {
        return $query->where('user_id', $userId);
    }
}

File: acapella_app/lib/shared/models/choir.dart
class Choir {
  final String id;
  final String name;
  final String location;
  final String bio;
  final String imageUrl;
  final int albumCount;
  final int trackCount;
  final bool isVerified;

  const Choir({
    required this.id,
    required this.name,
    required this.location,
    required this.bio,
    required this.imageUrl,
    required this.albumCount,
    required this.trackCount,
    this.isVerified = false,
  });
}
